src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -42,9 +42,22 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 		context.RegisterImplementationSourceOutput(bindings, (spc, binding) =>
 		{
 			var location = binding.SimpleLocation;
+			if (location == null)
+			{
+				throw new InvalidOperationException("Location cannot be null");
+			}
+
 			var fileName = $"{location.FilePath}-GeneratedBindingInterceptors-{location.Line}-{location.Column}.g.cs";
 			var sanitizedFileName = fileName.Replace('/', '-').Replace('\\', '-').Replace(':', '-');
-			var code = BindingCodeWriter.GenerateBinding(binding, (uint)Math.Abs(location.GetHashCode()));
+			var methodNamePrefix = binding.MethodType switch
+			{
+				InterceptedMethodType.SetBinding => "SetBinding",
+				InterceptedMethodType.Create => "Create",
+				_ => throw new NotSupportedException()
```

Comment:
Could we put an exception message that says "invalid MethodType"?

Reply:
this isn't part of this PR, BindingSourceGenerator is something different

Reply:
cc @simonrozsival